aikau/src/main/resources/alfresco/services/FileUploadService.js

```diff
@@ -0,0 +1,708 @@
+/**
+ * Copyright (C) 2005-2016 Alfresco Software Limited.
+ *
+ * This file is part of Alfresco
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * <p>This service can be used to control the uploading of content as well as
+ * the updating the content of existing nodes on an Alfresco Repository.</p>
+ * 
+ * @module alfresco/services/FileUploadService
+ * @extends module:alfresco/services/BaseService
```

Comment:
There seems to be a lot of duplication in this module from the UploadService. I think we can do a better job of abstraction of common functions here. I want to avoid any possibility of double maintenance.